### DISCUSSION TITLE
Cannot find type definition file for 'mime' 如何解决?

### DISCUSSION THREAD

**Discussion by ddrunLandalf** [midwayjs/midway#3739] [Q&A]
### Describe the problem（描述问题）

`
[ Midway ] Pre Process Force Exit.
error TS2688: Cannot find type definition file for 'mime'.
  The file is in the program because:
    Entry point for implicit type library 'mime'

Error! You can try adding the -V parameter for more information output.

TypeError: Cannot read properties of undefined (reading 'fileName')
    at BuildPlugin.emit (/www/wwwroot/ddrun/server/node_modules/@midwayjs/cli-plugin-build/src/index.ts:201:66)
    at CommandCore.execLifecycle (/www/wwwroot/ddrun/server/node_modules/@midwayjs/command-core/src/core.ts:216:11)
    at CommandCore.invoke (/www/wwwroot/ddrun/server/node_modules/@midwayjs/command-core/src/core.ts:182:5)
    at CLI.start (/www/wwwroot/ddrun/server/node_modules/@midwayjs/command-core/src/cli.ts:191:5)
`

### Midway Versions（Midway 版本）

"@midwayjs/bootstrap": "^3.5.1",
    "@midwayjs/bull": "3",
    "@midwayjs/cli": "^1.3.21",
    "@midwayjs/core": "^3.9.0",
    "@midwayjs/decorator": "^3.9.0",
    "@midwayjs/egg-ts-helper": "^1.0.1",
    "@midwayjs/logger": "^3.2.0",
    "@midwayjs/mock": "^3.5.1",
    "@midwayjs/orm": "^3.1.1",
    "@midwayjs/redis": "^3.5.1",
    "@midwayjs/socketio": "3",
    "@midwayjs/static-file": "3",
    "@midwayjs/task": "^3.5.1",
    "@midwayjs/upload": "^3.5.2",
    "@midwayjs/validate": "^3.5.2",
    "@midwayjs/web": "^3.5.1",
    "@midwayjs/ws": "^3.5.1",

**Comment by czy88840616** [comment]
我本地未出现，可以锁定到 @types/mime@1.3.5 试试

**Comment by LLL632648043** [comment]
你好，我直接根据官网文档安装并启动，也会出现和你一样的问题。
我切换node版本为20以上，使用pnpm命令运行则成功。以下是我的package.json内容：
{
  "name": "integration-weixin-work",
  "version": "1.0.0",
  "description": "",
  "private": true,
  "dependencies": {
    "@midwayjs/bootstrap": "^3.15.0",
    "@midwayjs/consul": "^3.15.0",
    "@midwayjs/core": "^3.15.0",
    "@midwayjs/decorator": "^3.15.0",
    "@midwayjs/http-proxy": "^3.15.0",
    "@midwayjs/jwt": "3",
    "@midwayjs/logger": "^3.3.0",
    "@midwayjs/redis": "3",
    "@midwayjs/static-file": "3",
    "@midwayjs/validate": "3.15.1",
    "@midwayjs/web": "^3.15.0",
    "address": "1.1.2",
    "egg": "^3.20.0",
    "flake-idgen": "^1.4.0",
    "ip-to-int": "^0.3.1",
    "skywalking-backend-js": "^0.7.0",
    "uuid": "^9.0.1"
  },
  "devDependencies": {
    "@midwayjs/egg-ts-helper": "^1.2.1",
    "@midwayjs/mock": "^3.15.0",
    "@types/consul": "^0.40.3",
    "@types/flake-idgen": "^0.1.34",
    "@types/jest": "^29.5.12",
    "@types/node": "^20.11.24",
    "@types/uuid": "^9.0.8",
    "cross-env": "^7.0.3",
    "egg-mock": "^5.10.9",
    "jest": "^29.7.0",
    "mwts": "^1.3.0",
    "mwtsc": "^1.7.1",
    "ts-jest": "^29.1.2",
    "ts-node": "^10.9.2",
    "tsconfig-paths": "^4.2.0",
    "typescript": "5.1.6"
  },
  "engines": {
    "node": ">=16"
  },
  "scripts": {
    "start": "NODE_ENV=production node ./bootstrap.js",
    "dev": "cross-env ets && cross-env NODE_ENV=local mwtsc --watch --run @midwayjs/mock/app.js",
    "test": "cross-env NODE_ENV=unittest NODE_OPTIONS=--require=ts-node/register jest",
    "cov": "cross-env NODE_OPTIONS=--require=ts-node/register jest --coverage",
    "lint": "mwts check",
    "lint:fix": "mwts fix",
    "ci": "npm run cov",
    "build": "mwtsc --cleanOutDir"
  },
  "repository": {
    "type": "git",
    "url": ""
  },
  "author": "",
  "license": "MIT"
}
